e2e/tests/prometheus_query.rs

```diff
@@ -0,0 +1,66 @@
+use chrono;
+use failure::{Fallible, ResultExt};
+use prometheus_query::v1::queries::{QueryData, QueryResult, VectorResult};
+use prometheus_query::v1::Client;
+#[test]
+fn query_prometheus() -> Fallible<()> {
+    let api_base = std::env::var("PROM_ENDPOINT").context("PROM_ENDPOINT not set")?;
+    let token = std::env::var("PROM_TOKEN").context("PROM_TOKEN not set")?;
+
+    let client = Client::builder()
+        .api_base(Some(api_base))
+        .access_token(Some(token))
+        .build()?;
+
+    let query = r#"up{job="apiserver"}"#;
+
+    let expected_json = r#"
+        {"status":"success","data":{"resultType":"vector","result":[{"metric":{"version":"4.0.0-0.3"},"value":[1552056334,"0.04081632653061224"]},{"metric":{"version":"4.0.0-0.alpha-2019-03-05-054505"},"value":[1552056334,"0.056451612903225805"]},{"metric":{"version":"4.0.0-0.6"},"value":[1552056334,"0.07692307692307693"]},{"metric":{"version":"4.0.0-0.7"},"value":[1552056334,"0.04519774011299435"]},{"metric":{"version":"4.0.0-0.5"},"value":[1552056334,"0.0625"]}]}}
+        "#;
+
+    let mut expected_result: Vec<VectorResult> =
+        match serde_json::from_str::<QueryResult>(expected_json)? {
+            QueryResult::Success(query_success) => match query_success.data() {
+                QueryData::Vector(vector) => vector.to_vec(),
+                _ => panic!("expected vector"),
+            },
+            _ => panic!("expected result"),
+        };
+
+    let mut runtime = commons::testing::init_runtime().unwrap();
+
+    let mut result: Vec<VectorResult> = match runtime.block_on(client.query(
+        query.to_string(),
+        Some(chrono::DateTime::from_utc(
+            chrono::NaiveDateTime::from_timestamp(1_552_056_334, 675 * 100),
+            chrono::Utc,
+        )),
+        None,
+    ))? {
+        QueryResult::Success(query_success) => match query_success.data() {
+            QueryData::Vector(vector) => vector.to_vec(),
+            _ => panic!("expected vector"),
+        },
+        _ => panic!("expected result"),
+    };
+
+    fn sort_by_version(a: &VectorResult, b: &VectorResult) -> std::cmp::Ordering {
+        let a = a.sample().clone();
+        let b = b.sample().clone();
+
+        if a < b {
+            std::cmp::Ordering::Less
+        } else if a > b {
+            std::cmp::Ordering::Greater
+        } else {
+            std::cmp::Ordering::Equal
+        }
+    }
+
+    expected_result.sort_by(sort_by_version);
+    result.sort_by(sort_by_version);
+
+    assert_eq!(expected_result, result);
```

Comment:
I suggest we skip this assertion, which leaves us with an integration tests checking for a successful query and deserialization, and failing if we somehow break that.

Reply:
Done. In any case this test is creating a different test binary - `prometheus-query-...` and it doesn't get copied to `e2e` container, so its effectively disabled

Reply:
Can we copy it or move it somewhere else so it's executed? The situation I want to prevent is where this codepath breaks untested, subsequently sabotages other tests, and we don't have a clue what's going on.